kola/tests/rpmostree/deployments.go

```diff
@@ -0,0 +1,302 @@
+	createBranch := "sudo ostree refs --create " + newBranch + " " + originalCsum
+	c.MustSSH(m, createBranch)
+
+	// make a commit to the new branch
+	createCommit := "sudo ostree commit -b " + newBranch + " --tree ref=" + originalCsum + " --add-metadata-string version=" + newVersion
+	newCommit := c.MustSSH(m, createCommit)
+
+	// use "rpm-ostree rebase" to get to the "new" commit
+	c.MustSSH(m, "sudo rpm-ostree rebase :"+newBranch)
+
+	// get latest rpm-ostree status output to check validity
+	postUpgradeStatus, err := getRpmOstreeStatusJSON(c, m)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// should have two deployments
+	if len(postUpgradeStatus.Deployments) != 2 {
+		c.Fatalf("Expected two deployments; found %d deployments", len(postUpgradeStatus.Deployments))
+	}
+
+	// reboot into new deployment
+	rebootErr := m.Reboot()
+	if rebootErr != nil {
+		c.Fatalf("Failed to reboot machine: %v", err)
+	}
+
+	// get latest rpm-ostree status output
+	postRebootStatus, err := getRpmOstreeStatusJSON(c, m)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// should still have 2 deployments
+	if len(postRebootStatus.Deployments) != 2 {
+		c.Fatalf("Expected two deployments; found %d deployment", len(postRebootStatus.Deployments))
+	}
+
+	// origin should be new branch
+	if postRebootStatus.Deployments[0].Origin != newBranch {
+		c.Fatalf(`New deployment origin is incorrect; expected %q, got %q`, newBranch, postRebootStatus.Deployments[0].Origin)
+	}
+
+	// new deployment should be booted
+	if !postRebootStatus.Deployments[0].Booted {
+		c.Fatalf("New deployment is not reporting as booted")
+	}
+
+	// checksum should be new commit
+	if postRebootStatus.Deployments[0].Checksum != string(newCommit) {
+		c.Fatalf(`New deployment checksum is incorrect; expected %q, got %q`, newCommit, postRebootStatus.Deployments[0].Checksum)
+	}
+
+	// version should be new version string
+	if postRebootStatus.Deployments[0].Version != newVersion {
+		c.Fatalf(`New deployment version is incorrect; expected %q, got %q`, newVersion, postRebootStatus.Deployments[0].Checksum)
+	}
+
+	// rollback to original deployment
+	c.MustSSH(m, "sudo rpm-ostree rollback")
+
+	newRebootErr := m.Reboot()
+	if newRebootErr != nil {
+		c.Fatalf("Failed to reboot machine: %v", err)
+	}
+
+	rollbackStatus, err := getRpmOstreeStatusJSON(c, m)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// still 2 deployments...
+	if len(rollbackStatus.Deployments) != 2 {
+		c.Fatalf("Expected two deployments; found %d deployments", len(rollbackStatus.Deployments))
+	}
+
+	// validate we are back to the original deployment by comparing the
+	// the two rpmOstreeDeployment structs
+	if !reflect.DeepEqual(originalStatus.Deployments[0], rollbackStatus.Deployments[0]) {
+		c.Fatalf(`Differences found in "rpm-ostree status"; original %v, current: %v`, originalStatus.Deployments[0], rollbackStatus.Deployments[0])
+	}
+
+	// cleanup our mess
+	cleanupErr := rpmOstreeCleanup(c, m)
```

Comment:
Unless we're wanting to test the cleanup we can drop this as the machine will just be terminated after the test is run (same for this in `rpmOstreeInstallUninstall`).

Reply:
So if I `kola run ... rhcos.rpmostree.*`, a new machine is started for each of the tests?

Reply:
That's correct. Only `NativeFuncs` (e.x.: `rhcos.basic`) in the same test are ran on the same machine.

Reply:
Huh...TIL.  :smile:

Reply:
I think I'll keep those in there just as a sanity test of `cleanup`